scripts/accumulator-server.py

```diff
@@ -46,7 +46,7 @@
 # * This script also depends on pyOpenSSL 19.0.0
 
 
-from OpenSSL import SSL
+# from OpenSSL import SSL
```

Comment:
This should be commented. Some .test run accumulator using HTTPS if I remember correctly.

Reply:
I commented out this line because the SSL interface is just used at line [586](https://github.com/telefonicaid/fiware-orion/pull/4035/files/c2ea3115b62363bf2cc8f3f39a81c99b6763e8a3#diff-eaa38239c9ce2beb0e36a37f9327707aea5fb10b1aa2d385a2c8eb94f86ca40aR586):

```
# context = SSL.Context(SSL.SSLv23_METHOD)
```

But as it's also commented out ... my pyflakes threw me a warning. 

But you are right, I'm not sure if ssl tests will pass with one of these two lienes commented out ... because after all, the SSL interface is not even used if line 586 is commented out.

Reply:
I have had a closer look to this and it seems due to an old version of the SSL stuff no longer used. In fact, this commented line and the commented block from L575 to L588 could be removed. You can do it in this PR or I can do it in a later one, as you prefer....

Reply:
what about the tuple `ssl_context`? I'd say we could also omit the `ssl_context`  argument for flask.

Reply:
> what about the tuple `context` at line 589? I'd say we could also omit the `ssl_context` argument for flask.
> 
> I mean, if we remove these lines ... we could drop the whole https support for the script.

Note that https support cannot be drop, as it is needed in some cases, pe. https://github.com/telefonicaid/fiware-orion/blob/master/test/functionalTest/cases/0706_direct_https_notifications/direct_https_notifications.test

Thus only the old implementation (in the commented lines) should be removed.

Reply:
well, apparently the code as its current state should work (according to [pyopenssl](https://www.pyopenssl.org/en/latest/api/ssl.html) docs). So probably uncommenting the lines will be enough.

Reply:
> So probably uncommenting the lines will be enough.

Actually, removing the currently commented lines.

Reply:
We could replace the used SSL context with a most recent one like `OpenSSL.SSL.TLSv1_2_METHOD`

When you say ...  'removing the currently commented lines' ... you mean to uncomment them? Or just delete them all?

Reply:
I refer to deleting this

```
 # from OpenSSL import SSL
```

and this

```
        """
        Commented lines correspond to the way of using context with pyOpenSSL 0.13.1
        (the one that comes with system Python in CentOS 7).
        We are using the new way (which is simpler) as we moved to pyOpenSSL 19.0.0.
        Referecence: Reference:
        http://stackoverflow.com/questions/28579142/attributeerror-context-object-has-no-attribute-wrap-socket/28590266
        We need to upgrade pyOpenSSL version due to problems of installing 0.13.1 inside virtualenv.
        Installing the module requires to compile some parts and this causes a conflict with base
        openssl devel libraries in CentOS 7 (solvable by hack, but we want to avoid it)
        """
        # context = SSL.Context(SSL.SSLv23_METHOD)
        # context.use_privatekey_file(key_file)
        # context.use_certificate_file(cert_file)
```

understanding that with this is enough to have HTTPS support in the script:

```
        context = (cert_file, key_file)
        app.run(host=host, port=port, debug=False, ssl_context=context)
```

However, it there are better alternatives to implement HTTPS support (the one you mention about `OpenSSL.SSL.TLSv1_2_METHOD`) they can be done too. However, maybe it's better in a next PR in order not introduce too much complexity in this one.

Reply:
Oh, I'm sorry, I'm not familiar enough with TLS layers ... but ... I'm not sure if Flask will accept plain cert and privkey files without setting them as a part of a proper context :thinking: 

Apparently we could use for this case the an _ad hoc_ certificate (https://blog.miguelgrinberg.com/post/running-your-flask-application-over-https -> __The simplest way to do it__ section)

Reply:
I would say that the current implementation of

```
        context = (cert_file, key_file)
        app.run(host=host, port=port, debug=False, ssl_context=context)
```

Corresponds with the case in the section "Self-Signed Certificates" in the document you cite.

Thus, it seems that keeping that would be enough.